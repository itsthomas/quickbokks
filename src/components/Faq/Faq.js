import React, { useState } from 'react';
import styled from 'styled-components';
import Accordion from '../Accordion/Accordion'

const Container = styled.div`
    padding: 40px 10px 0;
    grid-column-start: 1;
    grid-column-end: none;

    @media (min-width: 768px) {
        grid-column-start: 3;
        grid-column-end: 14;
    }

    @media (min-width: 960px) {
        grid-column-end: 13;
    }
`;

const Wrapper = styled.div`
`;

const Title = styled.h2`
    margin: 0;
    padding-left: 0;

    @media (min-width: 768px) {
        padding-left: 40px;
    }
`;

const FaqQuestions = styled.ul`
    margin-top: 20px;
    padding-left: 0;

    hr {
        border: none;
        height: 1px;
        background-color: #d8d6d6;
    }
`;

const Faq = ({faqs}) => {
    const [faqsList] = useState(Array.from(faqs.map((v, i) => { return {question:v.question, answer:v.answer,showPanel:true}})));

    return (
        <Container>
            <Wrapper>
                <Title>Frequently asked questions</Title>
                <FaqQuestions>
                    <hr></hr>
                    {
                        faqsList.map(
                            (item, index) => 
                            <Accordion key={index} data={item}/>
                        )
                    }
                </FaqQuestions>
            </Wrapper>
        </Container>
    )
};

export default Faq;
