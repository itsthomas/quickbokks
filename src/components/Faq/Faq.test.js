import { render } from '@testing-library/react';
import Faq from './Faq'

describe("The Faq component", () => {
    const faqs = [
        {
            question: 'Sed erat dolor, pellentesque eu?',
            answer: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.'
        },
        {
            question: 'Praesent soliicitudin mauris a mattis?',
            answer: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.'
        }
    ];

    test("should render", () => {
      const { asFragment } = render(<Faq faqs={faqs}/>);
      const html = asFragment();
      expect(html).toMatchSnapshot();
    });
});
