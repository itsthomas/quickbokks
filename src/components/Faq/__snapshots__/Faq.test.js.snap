// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Faq component should render 1`] = `
<DocumentFragment>
  <div
    class="sc-dlfnbm fiLKhl"
  >
    <div
      class="sc-hKgILt chrFSH"
    >
      <h2
        class="sc-eCssSg ckLFtq"
      >
        Frequently asked questions
      </h2>
      <ul
        class="sc-jSgupP fgTMIE"
      >
        <hr />
        <li
          class="sc-gsTCUz efUBJI"
        >
          Sed erat dolor, pellentesque eu?
        </li>
        <hr />
        <li
          class="sc-gsTCUz efUBJI"
        >
          Praesent soliicitudin mauris a mattis?
        </li>
        <hr />
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
